### PR TITLE
docs: support copying token

### PR DESCRIPTION
**Pull request by TinsFox**
For ease of use, I have supported the copying of the current color token in the document.

image

### CODE REVIEW

File: docs/app/layout.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const imgUrl = 'https://repository-images.githubusercontent.com/453236442/aa0db6
 export default function Layout({ children }) {
   return (
     <html>
-      <body>
+      <body suppressHydrationWarning>
```
Comment (huozhi) [inline docs/app/layout.js:8]: Why we add this, we should need to suppress warning in the boday
```suggestion
      <body>
```

File: docs/app/live-editor.js [hunk 2]
```diff
@@ -178,6 +190,8 @@ export default function LiveEditor() {
                     })
                   }}
                 />
+                <span>{colorPlateColors[tokenTypeName]}</span>
+                {/* <CopyButton codeSnippet={`${tokenTypeName}--sh-: ${colorPlateColors[tokenTypeName]}`}/> */}
```
Comment (huozhi) [inline docs/app/live-editor.js:194]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -26,4 +27,4 @@
     "vitest": "^1.2.2"
   },
   "packageManager": "pnpm@8.7.1"
-}
+}
```
Comment (huozhi) [inline package.json:30]: let's revert the unrelated changes and keep the file as it was

File: docs/app/live-editor.js [hunk 4]
```diff
@@ -126,6 +127,13 @@ export default function LiveEditor() {
   }))
   const debouncedTokenize = debouncedTokenizeRef.current
 
+  const customizableColorsString=useMemo(()=>{
```
Comment (huozhi) [inline docs/app/live-editor.js:130]: need to prettier the changes

File: docs/app/components/copy-button.css [hunk 5]
```diff
@@ -0,0 +1,11 @@
+.copy-button{
+  background-color:transparent
+}
+
+button{
```
Comment (huozhi) [inline docs/app/components/copy-button.css:5]: let's use class name instead of html tag selector, also add an empty line at the end

**Review by huozhi — CHANGES_REQUESTED** [review]
Thanks for the PR, nice feature!

Notice there're also some extra spaces on the top of color plate, could you fix them aligning with current version? Thanks

**Comment by TinsFox** [comment]
Oh, thank you for offering so many suggestions, I will revise it!

File: docs/app/layout.js (hunk 1 above)
Comment (TinsFox) [reply docs/app/layout.js:8]: I added this because I created an empty Next.js project using `npx create-next-app@latest`, and even after removing most of the content, I still encountered the following error. I believe this is a bug in Next.js, which can lead to our incorrect judgment. Therefore, I have added a prohibition on errors here.

image

image

In fact, I'm not very clear whether `suppressHydrationWarning` should really be added here. If you have any knowledge in this area, I look forward to your sharing ❤️

File: docs/app/live-editor.js (hunk 4 above)
Comment (TinsFox) [reply docs/app/live-editor.js:130]: Can you share your Prettier configuration? My configuration should be different from yours.

Comment (huozhi) [reply docs/app/live-editor.js:130]: Or it’s good to just leave the space between the operators like others. I didn’t prettify it but prefer the consistent code style

**Comment by TinsFox** [comment]
> Notice there're also some extra spaces on the top of color plate, could you fix them aligning with current version?

Are you referring to this gap?

image

File: docs/app/live-editor.js (hunk 4 above)
Comment (TinsFox) [reply docs/app/live-editor.js:130]: I did that just now, do you see if there are any other issues?

**Comment by huozhi** [comment]
If you compare with production, the code and "color plate" are having the same padding/marging to the top

image

File: docs/app/layout.js (hunk 1 above)
Comment (TinsFox) [reply docs/app/layout.js:8]: Thank you for sharing, I suspect it might be an issue with my Chrome extension. I will roll back this change later.

**Comment by TinsFox** [comment]
> If you compare with production, the code and "color plate" are having the same padding/marging to the top

I get what you mean.

**Comment by TinsFox** [comment]
@huozhi I just fixed the color plate issue. If you have time, you can check it out again ❤️

**Review by huozhi — APPROVED** [review]
I've fixed the formatting issues, thanks for the PR!
